linq4j/src/main/java/org/apache/calcite/linq4j/tree/ExpressionWriter.java

```diff
@@ -32,15 +32,25 @@ class ExpressionWriter {
 
   private final Spacer spacer = new Spacer(0);
   private final StringBuilder buf = new StringBuilder();
-  private boolean indentPending;
   private final boolean generics;
+  private final int methodSplittingThreshold;
+  private boolean indentPending;
 
   ExpressionWriter() {
-    this(true);
+    this(true, 0);
   }
 
-  ExpressionWriter(boolean generics) {
+  ExpressionWriter(boolean generics, int methodSplittingThreshold) {
     this.generics = generics;
+    this.methodSplittingThreshold = methodSplittingThreshold;
+  }
+
+  public ExpressionWriter duplicateState() {
```

Comment:
public method needs javadoc

Reply:
I made this package-private rather than adding javadoc -- it really seems like most of the methods in ExpressionWriter can be package-private.